src/neuroglancer/sliceview/compresso/index.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright William Silvermsith
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TypedArray} from 'neuroglancer/util/array';
+
+// @ts-ignore
+import createCompressoModule from './compresso.js';
+
+const compressoModulePromise : any = createCompressoModule();
+
+// not a full implementation of read header, just the parts we need
+function read_header(buffer: Uint8Array) : Map<string, number> {
+  // check for header "cpso"
+  const magic = (
+       buffer[0] === 'c'.charCodeAt(0) && buffer[1] === 'p'.charCodeAt(0)
+    && buffer[2] === 's'.charCodeAt(0) && buffer[3] === 'o'.charCodeAt(0)
+  );
+  if (!magic) {
+    throw new Error("compresso: didn't match magic numbers")
+  }
+  const format = buffer[4];
+  if (format !== 0) {
+    throw new Error("compresso: didn't match format version")
+  }
+
+  let u16 = (lower : number, upper: number) => { return (lower|0) + (upper << 8) };
+
+  const data_width = buffer[5];
+  const sx = u16(buffer[6], buffer[7]);
+  const sy = u16(buffer[8], buffer[9]);
+  const sz = u16(buffer[10], buffer[11]);
+
+  let map = new Map<string,number>();
```

Comment:
I think it would be better to use an object rather than a map for this.

Reply:
I ordinarily would, but I'm not sure how to manage the types with an object. Typescript n00b problem...

Reply:
You can declare the return type as something like

```typescript
{sx: number, sy: number, sz: number, dataWidth: number}
```

and then in the function body use:

```typescript
return {sx, sy, sz, dataWidth};
```